### ISSUE TITLE
Handle tasks spawned by a task pool in SpecializedPools

### ISSUE THREAD

**Issue by daniel-zullo** [sociomantic-tsunami/ocean#643]
The `SpecializedPools`  is not used if the task is spawned/started from another object rather than a plain task, for instance, spawned by a TaskPool. 
`SpecializedPools` works fine if a plain task is directly scheduled.

```D
unittest
{
    const NUM_START_CALLS = 100;

    static class SubTask : Task
    {
        static int value;

        public void copyArguments ()
        {
        }

        override public void run ()
        {
            // only works using the specialized pool
            long[5_000] big_array;

            .wait(20); // so that the task gets suspended

            SubTask.value++;
        }
    }

    class PoolTask : Task
    {
        TaskPool!(SubTask) my_task_pool;

        public this ()
        {
            this.my_task_pool = new TaskPool!(SubTask);
        }

        public void run ()
        {
            for (int i = 0; i < NUM_START_CALLS; i++)
                this.my_task_pool.start();

            this.my_task_pool.awaitRunningTasks();
        }
    }

    class PlainTask : Task
    {
        SubTask[NUM_START_CALLS] my_task_array;

        public this ()
        {
            foreach (ref task; this.my_task_array)
                task = new SubTask();
        }

        public void run ()
        {
            foreach (ref task; this.my_task_array)
            {
                theScheduler.schedule(task);
            }
        }
    }

    SchedulerConfiguration config;
    config.task_queue_limit = 200;
    config.worker_fiber_limit = 10;
    config.suspended_task_limit = 20;
    config.worker_fiber_stack_size = 32_768;

    config.specialized_pools = [
        config.PoolDescription(SubTask.classinfo.name, 65_536)
    ];

    initScheduler(config);
    theScheduler.schedule(new PlainTask()); // Works
    theScheduler.schedule(new PoolTask());  // Fails: segfault
    theScheduler.eventLoop();

    test!("==")(SubTask.value, NUM_START_CALLS * 2);
}
```

**Comment by daniel-zullo** [comment]
This bug/issue is solved by updating the method `findPool()` to: 

```diff
diff --git a/src/ocean/task/internal/SpecializedPools.d b/src/ocean/task/internal/SpecializedPools.d
index 2dbdd71..12c3c08 100644
--- a/src/ocean/task/internal/SpecializedPools.d
+++ b/src/ocean/task/internal/SpecializedPools.d
@@ -14,10 +14,12 @@
 
 module ocean.task.internal.SpecializedPools;
 
+import ocean.meta.codegen.CTFE;
 import ocean.meta.types.Qualifiers;
 import ocean.task.internal.FiberPoolEager;
 import ocean.task.IScheduler;
 import ocean.task.Task;
+import ocean.core.array.Search;
 import ocean.core.Verify;
 import ocean.core.Enforce;
 import ocean.core.Optional;
@@ -99,8 +101,14 @@ public class SpecializedPools
     {
         foreach (meta; this.mapping)
         {
-            if (meta.task_name == task)
+            auto specialized_name = stripQualifiedPrefix(meta.task_name);
+            auto is_from_pool = task.contains("OwnedTask");
+
+            if (meta.task_name == task ||
+                (is_from_pool && task.contains(specialized_name)))
+            {
                 return optional(meta);
+            }
         }
 
         return Optional!(SpecializedPool).undefined;
```

If you are happy with this kind of hacky solution I'll push a patch including a unit test. A better solution would require breaking changes in the API

**Comment by daniel-zullo** [comment]
Addressed by #645
